### ISSUE TITLE
Improve workflow for maintainers

### ISSUE THREAD

**Issue by ryanofsky** [PierreRochard/bitcoin-acks#75]
This is maybe an overbroad issue, but it seems Bitcoin maintainers are currently struggling to keep track of open bitcoin PRs. From today's IRC meeting: "I can't manually, humanly, handle 200+ PRs, sorry" (https://botbot.me/freenode/bitcoin-core-dev/msg/104301215/).

I'm wondering if it would be possible to add features to bitcoin-acks that could improve maintainer workflow. Two examples that come to mind:

- Maybe bitcoinacks could detect a #needs-maintainer-attention or similar tag in comments, and detect maintainer responses. This way people could flag PRs that seem ready for merging or need other input, and maintainers could see the PRs that need actually need their attention, while giving other PRs less priority.

- Maybe bitcoinacks could give maintainers a snooze feature, or a way of assigning their own labels or priorities to PRs. For example if @laanwj wants to only consider merging refactoring changes once a month, and not see a list of PRs filled with refactoring changes all the time, maybe bitcoinacks could provide his desired view.

These are just examples, and actual maintainers could probably come up with better suggestions, but it seems to me bitcoin-acks could be pretty useful as a tool for maintainers to keep with PRs, and it might be worth investigating if any new features could help.

**Comment by PierreRochard** [comment]
> First question here for Pierre is just whether you think features targeted at maintainers would be acceptable or make sense as part of bitcoinacks.com. 

Yes, and if maintainer decision-making is the bottleneck, features that help maintainers are my number one priority for bitcoinacks.com

I think both of the examples given are going in the right direction, and I too would be interested in hearing from the maintainers as well as contributors interested in taking on a maintainer role (formally or de facto "code owner"). 

I won't be at Scaling Bitcoin next month but I hope scaling the Bitcoin Core contribution process gets brought up, and if people there have ideas relevant to bitcoinacks.com I'm all ears.

**Comment by laanwj** [comment]
FWIW, Bitcoinacks is already a giant help in sorting out the endless list of issues. For example by being able to hide PRs that need rebase.

The "hide these PRs until the end of the month" or so option sounds like a good idea to me, as well. 

Would be good to show a list of good candidates for review/merging without getting stuck in a forest of random refactoring/code style PRs. These seem easy, "oh, it'll only take a few minutes", but because they're overrepresented, and do (sometimes) tend to cause a lot of discussion, all in all they do take a lot of time to consider.

**Comment by PierreRochard** [comment]
> The "hide these PRs until the end of the month" or so option sounds like a good idea to me, as well.

I think this would require authentication so that different users can hide/snooze different PRs, I already have a prototype integration with GitHub oauth in a branch but I was concerned that it would be bad for user privacy (as in, I can see everything in the database, though anyone can run it locally to avoid that), I don't know if contributors and maintainers share that privacy concern in this context.



> Would be good to show a list of good candidates for review/merging

I see three ways of doing this:

1. automated heuristic (for example >3 (u)tack reviews, two of which from the top 30 contributors over the past 12 months, integration tests passing, no rebase needed, does not conflict with a high priority PR, etcetera)

2. Manually flagged in the PR with a comment like @ryanofsky's suggested

3. Manually flagged on bitcoinacks.com

Option (3) would be the easiest for me to implement, but would also require oauth integration to prevent abuse / limit the flagging to contributors or the PR author. We could also have both automated and manual flagging.

**Comment by PierreRochard** [comment]
As a first step I added "Ready for Review" and "Ready for Maintainer" buttons at the top. The latter narrows the list down to 40 pull requests

**Comment by ryanofsky** [comment]
@MarcoFalke and @laanwj I'd be curious if the new features Pierre added are useful to you.

I guess we are already closing prs due to lack of priority, (https://github.com/bitcoin/bitcoin/pull/14211#issuecomment-423324019, https://github.com/bitcoin/bitcoin/pull/14217), but still it would be nice to know if there's anything that could be done to manage priority better.

Another thing I was wondering about was whether the Pierre's "Ready for Maintainer" tracking could also be used for email filtering. E.g. if bitcoinacks could provide a REST API or something that would let you prioritize notifications and avoid spam from refactoring prs if you want to only want to check them occasionally.
